### PR TITLE
Yen's answers to diagnostic day 1_SecondWeek

### PR DESCRIPTION
**Pull request by s0phi4**
Comfort: 3
Clarity: 3 I still have some confusion creating functions... 

I used my notes and the code created in the lab last week.

### CODE REVIEW

File: lib/diagnostic.js [hunk 1]
```diff
@@ -16,8 +16,14 @@
...
-// Answer Starts Here
-//
+// Description: Calendar tracks bills to be paid
+//Entity: Calendar
+//Attributes: Year, Months, Days
+//Methods: save, alert
+
+// Entity: bill
+// Attributes: Company, amount
+// Methods:
```
Comment (bernardlee) [inline lib/diagnostic.js:26]: ```payBill()``` is an example of a possible method.

File: lib/diagnostic.js [hunk 2]
```diff
@@ -38,14 +46,35 @@ let owner = null;
 // Methods: vocalize
 
 // Replace null with your answer
-let pet = null;
+let pet = {name: 'Boozer',
+          speciess: 'Dog',
```
Comment (bernardlee) [inline lib/diagnostic.js:50]: ```speciess``` should be ```species```.  Run ```grunt test``` and watch for typos.

File: lib/diagnostic.js [hunk 3]
```diff
@@ -38,14 +46,35 @@ let owner = null;
...
+          breed: 'Labrador',
+          vocalization: 'BuUZeR',
+
+          vocalize: function (name){
+            if (name === this.name){
+            return pet.vocalization;
+          }
+          }
+
+};
```
Comment (BenGitsCode) [inline lib/diagnostic.js:60]: It looks like you're trying to make vocalize run only if you call the pet by it's name? We're looking for something more simple:

```js
let pet = {
  name: 'Beau',
  species: 'Dog',
  breed: 'Lab',
  vocalization: 'bark',
  vocalize: function () {
    return this.vocalization;
  }
};
```

File: lib/diagnostic.js [hunk 4]
```diff
@@ -38,14 +46,35 @@ let owner = null;
...
+const Pet = function (name, species, breed, vocalization) {
+  this.name = name;
+  this.species = species;
+  this.breed = breed;
+  this.vocalization = vocalization;
+};
+
+Pet.prototype.vocalize = function () {
+  let name = this.name;
+  if (name === this.name){
```
Comment (bernardlee) [inline lib/diagnostic.js:75]: This is unnecessary.  In the line above, you set ```name``` to ```this.name```, so ```name === this.name``` will always be true.

**Comment by bernardlee** [comment]
5/5

File: lib/diagnostic.js [hunk 5]
```diff
@@ -38,14 +46,35 @@ let owner = null;
...
+  return pet.vocalization;
+}
 };
```
Comment (BenGitsCode) [inline lib/diagnostic.js:78]: Overthinking it from the looks of it.
```js
const vocalize = function () {
  return 'bark';
};

Pet.prototype.vocalize = vocalize;
```

**Pull request closed without merging**